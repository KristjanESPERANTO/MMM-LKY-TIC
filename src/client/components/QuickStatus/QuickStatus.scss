@import "../../styles/common";

/** @define quickstatus */

.mmm-lky-tic-wrapper {
  .quick-status {
    display: flex;
    flex-direction: column;
    padding-bottom: 0.5em;
    margin-bottom: 0.5em;
    border-bottom: 1px solid var(--color-text-dimmed);

    &__items {
      display: flex;
      flex-flow: row wrap;
    }

    &__link-item,
    &__supply-item,
    &__fare-period-item {
      margin: 0 0.5em;
    }

    &__link-picto.not-connected,
    &__supply-picto.supply-critical {
      animation: blinker 1s linear infinite;
    }

    &__fare-period-item {
      font-size: 0.8em;
    }
  }
}
