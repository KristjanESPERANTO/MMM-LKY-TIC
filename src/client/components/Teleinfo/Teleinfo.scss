@import "../../styles/common";

/** @define teleinfo */

.mmm-lky-tic-wrapper {
  .teleinfo {
    display: flex;
    flex-direction: column;
    font-size: 0.9em;

    &__no-data {
      font-style: italic;
    }

    &__power-value {
      font-size: var(--font-size-large);
    }

    &__power-unit,
    &__power-stats-unit,
    &__intensity-unit,
    &__supplied-unit,
    &__costs-unit {
      color: $highlighted-text;
    }

    &__power-unit {
      font-size: var(--font-size-medium);
    }

    &__power-stats-unit,
    &__intensity-unit,
    &__supplied-unit,
    &__costs-unit {
      font-size: var(--font-size-xsmall);
    }

    &__power {
      justify-content: center;
    }

    &__power,
    &__power-stats-group,
    &__over-intensity,
    &__intensity,
    &__costs,
    &__supplied,
    &__supplied-detail {
      display: flex;
      flex-direction: row;
      align-items: baseline;
    }

    &__supplied-detail {
      justify-content: right;
    }

    &__supplied-detail-items,
    &__power-stats {
      display: flex;
      flex-direction: column;
    }

    &__power-stats,
    &__supplied,
    &__costs {
      font-size: var(--font-size-small);
    }

    &__supplied-detail,
    &__over-intensity {
      font-size: var(--font-size-xsmall);
    }

    &__intensity,
    &__over-intensity,
    &__costs,
    &__supplied-detail,
    &__dates-start-last {
      margin-bottom: 0.5em;
    }

    &__intensity.with-overflow {
      margin-bottom: 0;
    }

    &__dates-start-last {
      font-style: italic;
      font-size: 0.8em;
    }

    &__power-value,
    &__power-stats-value,
    &__intensity-value,
    &__costs-value,
    &__supplied-value,
    &__dates-start-last-value {
      color: $highlighted-text;
    }

    &__intensity,
    &__over-intensity,
    &__dates {
      margin-left: auto;
    }

    &__over-intensity {
      animation: blinker 2s linear infinite;
      color: $critical-text;
    }

    &__power-stats-label,
    &__intensity-label,
    &__over-intensity-label,
    &__supplied-label,
    &__costs-label,
    &__dates-start-last-label {
      margin-right: 0.5em;
    }

    &__instant-section,
    &__dates-section,
    &__costs-section {
      display: flex;
      flex-direction: column;

      &::after {
        content: "";
        display: block;
        border-bottom: 1px dashed var(--color-text-dimmed);
      }
    }

    &__costs-section,
    &__dates-section {
      padding-top: 0.5em;
    }
  }
}
