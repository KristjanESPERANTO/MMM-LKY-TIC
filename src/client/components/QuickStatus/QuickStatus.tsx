import { faCircleQuestion } from '@fortawesome/free-regular-svg-icons';
import { faBoltLightning, faLeaf, faPlugCircleBolt, faPlugCircleExclamation } from '@fortawesome/free-solid-svg-icons';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import classNames from 'classnames';
import timeDifferenceInSeconds from 'date-fns/differenceInSeconds';
import { FunctionComponent } from 'react';
import { TeleInfo } from '../../../shared/domain/teleinfo';

import './QuickStatus.scss';

export interface QuickStatusProps {
  data?: TeleInfo;
  hearbeatTs?: number;
}

/**
 * Teleinfo quick status bar
 */
const QuickStatus: FunctionComponent<QuickStatusProps> = (props: QuickStatusProps) => {
  function renderItems() {
    return (
      <>
        <li key="qs-link" className="quick-status__link-item">
          {renderLinkItem()}
        </li>
        <li key="qs-supply" className="quick-status__supply-item">
          {renderSupplyItem()}
        </li>
      </>
    );
  }

  function renderLinkItem() {
    const options = {
      color: 'red',
      icon: faPlugCircleExclamation,
      isCritical: false,
    };

    if(props.data) {
      const lastReceivedTime = props.data.meta?.lastUpdateTimestamp || 0;
      const heartbeatTime = props.hearbeatTs || 0;
      const idleDuration = timeDifferenceInSeconds(heartbeatTime, lastReceivedTime);
      
      if (idleDuration > 30) {
        options.color = 'red';
        options.icon = faPlugCircleExclamation;
        options.isCritical = true;
      } else if (idleDuration > 10) {
        options.color = 'orange';
        options.icon = faPlugCircleExclamation;
      } else if (idleDuration > 5) {
        options.color = 'yellow';
        options.icon = faPlugCircleExclamation;
      } else {
        options.color = 'green';
        options.icon = faPlugCircleBolt;  
      }
   }

    const itemClassName = classNames('quick-status__link-picto', {
      'not-connected': options.isCritical,
    });
    return (
      <>
        <FontAwesomeIcon className={itemClassName} color={options.color} icon={options.icon} />
      </>
    );
  }

  function renderSupplyItem() {
    const options = {
      color: 'white',
      icon: faCircleQuestion,
      isCritical: false,
    };

    const instantIntensity = props.data?.instantIntensity || 0;
    const maxIntensity = props.data?.subscribedIntensity || Number.POSITIVE_INFINITY;

    if (props.data?.subscribedPowerOverflowWarning) {
      options.color = 'red';
      options.icon = faBoltLightning;
      options.isCritical = true;
    } else if (instantIntensity > maxIntensity * 5 / 6) {
      options.color = 'red';
      options.icon = faBoltLightning;
    } else if (instantIntensity > maxIntensity * 2 / 3) {
      options.color = 'orange';
      options.icon = faBoltLightning;
    } else if (instantIntensity > maxIntensity / 3) {
      options.color = 'yellow';
      options.icon = faBoltLightning;
    } else if (props.data) {
      options.color = 'green';
      options.icon = faLeaf;
    }

    const itemClassName = classNames('quick-status__supply-picto', { 'supply-critical': options.isCritical });
    return (
      <>
        <FontAwesomeIcon className={itemClassName} color={options.color} icon={options.icon} />
      </>
    );
  }

  return (
    <div className="quick-status">
      <ul className="quick-status__items">
        {renderItems()}
      </ul>
    </div>
  )
};

export default QuickStatus;
