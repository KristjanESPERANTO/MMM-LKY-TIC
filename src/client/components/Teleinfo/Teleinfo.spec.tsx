// Must be located BEFORE the imports
const mockFormatDate = jest.fn((ts: number, format: string) => `${format}(${ts})`);
const mockUseWithNotifications = jest.fn();

import React from 'react';
import renderer from 'react-test-renderer';
import { QuickStatusProps } from '../QuickStatus/QuickStatus';
import Teleinfo, { NotificationData } from './Teleinfo';

jest.mock('date-fns/format', () => mockFormatDate);

const mockUseContext = jest.spyOn(React, 'useContext');

jest.mock('../../hooks/with-notifications/with-notifications', () => mockUseWithNotifications);

jest.mock('../QuickStatus/QuickStatus', () => (props: QuickStatusProps) => <div {...props}>QuickStatus component</div>);

describe('Teleinfo component', () => {
  const defaultProps = {};
  const fullNotifData: NotificationData = {
    data_TELEINFO: {
      meta: {
        firstDataTimestamp: 1674565891995,
        lastUpdateTimestamp: 1674818126026,
        unresolvedGroups: {},
      },
      apparentPower: 250,
      chosenFareOption: 'BASE',
      estimatedPower: 225,
      estimatedPrices:{
        currentDay: 5.567,
        currentMonth: 55,
        total: 500
      },
      instantIntensity: 1,
      statistics: {
        instantIntensity: {},
        instantPower: {},
      },
      suppliedPower: {
        currentDay: [1, 2],
        currentMonth: [10, 20],
        total: [100, 200],
      },
    },
    data_TELEINFO_HEARTBEAT: {
      ts: 1674818126028
    },
  };

  beforeEach(() => {
    mockUseWithNotifications.mockReset();
    mockUseContext.mockReset();
    mockUseContext.mockReturnValue({
      currencySymbol: 'â‚¬',
    });
  });

  it('should render correctly without notif data', () => {
    // given
    mockUseWithNotifications.mockReturnValue({});

    // when
    const tree = renderer.create(<Teleinfo {...defaultProps} />).toJSON();

    // then
    expect(tree).toMatchSnapshot();
    expect(mockUseWithNotifications).toHaveBeenCalledWith(['TELEINFO', 'TELEINFO_HEARTBEAT']);
  });

  it('should render correctly with notif data', () => {
    // given
    mockUseWithNotifications.mockReturnValue(fullNotifData)

    // when
    const tree = renderer
      .create(<Teleinfo {...defaultProps} />)
      .toJSON();

    // then
    expect(tree).toMatchSnapshot();
  });

  it('should render correctly with notif data and power overuse', () => {
    // given
    const dataWithPowerOveruse: NotificationData = {
      ...fullNotifData,
      data_TELEINFO: {
        ...fullNotifData.data_TELEINFO,
        statistics: {
          ...fullNotifData.data_TELEINFO?.statistics,
        },
        subscribedPowerOverflowWarning: 1,
      },
    };
    mockUseWithNotifications.mockReturnValue(dataWithPowerOveruse)

    // when
    const tree = renderer
      .create(<Teleinfo {...defaultProps} />)
      .toJSON();

    // then
    expect(tree).toMatchSnapshot();
  });  
  
  it('should render correctly with notif data and power stats', () => {
    // given
    const dataWithPowerOveruse: NotificationData = {
      ...fullNotifData,
      data_TELEINFO: {
        ...fullNotifData.data_TELEINFO,
        statistics: {
          instantPower: {
            currentDay: {
              min: 1,
              max: 155,
            },
            currentMonth: {
              min: 1.5,
              max: 200,
            },
            overall: {
              min: 1.55,
              max: 500,
            }
          },
        },
        subscribedPowerOverflowWarning: 1,
      },
    };
    mockUseWithNotifications.mockReturnValue(dataWithPowerOveruse)

    // when
    const tree = renderer
      .create(<Teleinfo {...defaultProps} />)
      .toJSON();

    // then
    expect(tree).toMatchSnapshot();
  });

  it('should render correctly with notif data but without supplied power information', () => {
    // given
    const dataWithoutSuppliedPowerInfo: NotificationData = {
      ...fullNotifData,
      data_TELEINFO: {
        ...fullNotifData.data_TELEINFO,
        statistics: {
          ...fullNotifData.data_TELEINFO?.statistics,
        },
        suppliedPower: {},
      },
    };
    mockUseWithNotifications.mockReturnValue(dataWithoutSuppliedPowerInfo)

    // when
    const tree = renderer.create(<Teleinfo {...defaultProps} />).toJSON();

    // then
    expect(tree).toMatchSnapshot();
  });
});
