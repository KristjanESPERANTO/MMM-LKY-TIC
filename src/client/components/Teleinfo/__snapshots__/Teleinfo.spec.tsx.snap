// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Teleinfo component should render correctly with notif data 1`] = `
<div
  className="teleinfo"
>
  <section
    className="teleinfo__quick-status"
  >
    <div
      data={
        {
          "apparentPower": 250,
          "chosenFareOption": "BASE",
          "estimatedPower": 225,
          "estimatedPrices": {
            "currentDay": 5.567,
            "currentMonth": 55,
            "total": 500,
          },
          "instantIntensity": 1,
          "meta": {
            "firstDataTimestamp": 1674565891995,
            "lastUpdateTimestamp": 1674818126026,
            "unresolvedGroups": {},
          },
          "statistics": {
            "instantIntensity": {},
            "instantPower": {},
          },
          "suppliedPower": {
            "currentDay": [
              1,
              2,
            ],
            "currentMonth": [
              10,
              20,
            ],
            "total": [
              100,
              200,
            ],
          },
        }
      }
      hearbeatTs={1674818126028}
    >
      QuickStatus component
    </div>
  </section>
  <section
    className="teleinfo__instant-section"
  >
    <p
      className="teleinfo__power"
    >
      <span
        className="teleinfo__power-value"
      >
        250
      </span>
      <span
        className="teleinfo__power-unit"
      >
        VA
      </span>
      <span
        className="teleinfo__power-value"
      >
        ~
        225
      </span>
      <span
        className="teleinfo__power-unit"
      >
        W
      </span>
    </p>
    <p
      className="teleinfo__power-stats"
    >
      <p
        className="teleinfo__power-stats-group"
      >
        <span
          className="teleinfo__power-stats-label"
        >
          Min (today/month/overall):
        </span>
        <span
          className="teleinfo__power-stats-value"
        >
          ...
        </span>
        /
        <span
          className="teleinfo__power-stats-value"
        >
          ...
        </span>
        /
        <span
          className="teleinfo__power-stats-value"
        >
          ...
        </span>
        <span
          className="teleinfo__power-stats-unit"
        >
          VA
        </span>
      </p>
      <p
        className="teleinfo__power-stats-group"
      >
        <span
          className="teleinfo__power-stats-label"
        >
          Max:
        </span>
        <span
          className="teleinfo__power-stats-value"
        >
          ...
        </span>
        /
        <span
          className="teleinfo__power-stats-value"
        >
          ...
        </span>
        /
        <span
          className="teleinfo__power-stats-value"
        >
          ...
        </span>
        <span
          className="teleinfo__power-stats-unit"
        >
          VA
        </span>
      </p>
    </p>
    <p
      className="teleinfo__intensity"
    >
      <span
        className="teleinfo__intensity-label"
      >
        Intensity:
      </span>
      <span
        className="teleinfo__intensity-value"
      >
        1
      </span>
      <span
        className="teleinfo__intensity-unit"
      >
        A
      </span>
    </p>
    <p
      className="teleinfo__supplied"
    >
      <span
        className="teleinfo__supplied-label"
      >
        Supplied (today/month/total):
      </span>
      <span
        className="teleinfo__supplied-value"
      >
        3
      </span>
      /
      <span
        className="teleinfo__supplied-value"
      >
        30
      </span>
      /
      <span
        className="teleinfo__supplied-value"
      >
        300
      </span>
      <span
        className="teleinfo__supplied-unit"
      >
        Wh
      </span>
    </p>
    <p
      className="teleinfo__supplied-detail"
    >
      <ul
        className="teleinfo__supplied-detail-items"
      >
        <li
          className="teleinfo__supplied-detail-item"
        >
          <span
            className="teleinfo__supplied-label"
          >
            base
            :
          </span>
          <span
            className="teleinfo__supplied-value"
          >
            1
          </span>
          /
          <span
            className="teleinfo__supplied-value"
          >
            10
          </span>
          /
          <span
            className="teleinfo__supplied-value"
          >
            100
          </span>
          <span
            className="teleinfo__supplied-unit"
          >
            Wh
          </span>
        </li>
        <li
          className="teleinfo__supplied-detail-item"
        >
          <span
            className="teleinfo__supplied-label"
          >
            :
          </span>
          <span
            className="teleinfo__supplied-value"
          >
            2
          </span>
          /
          <span
            className="teleinfo__supplied-value"
          >
            20
          </span>
          /
          <span
            className="teleinfo__supplied-value"
          >
            200
          </span>
          <span
            className="teleinfo__supplied-unit"
          >
            Wh
          </span>
        </li>
      </ul>
    </p>
  </section>
  <section
    className="teleinfo__costs-section"
  >
    <p
      className="teleinfo__costs"
    >
      <span
        className="teleinfo__costs-label"
      >
        Costs (today/month/total):
      </span>
      <span
        className="teleinfo__costs-value"
      >
        ~
        5.57
      </span>
      /
      <span
        className="teleinfo__costs-value"
      >
        55
      </span>
      /
      <span
        className="teleinfo__costs-value"
      >
        500
      </span>
      <span
        className="teleinfo__costs-unit"
      >
        €
      </span>
    </p>
    <p
      className="teleinfo__costs-detail"
    >
      <ul
        className="teleinfo__costs-detail-items"
      />
    </p>
  </section>
  <section
    className="teleinfo__dates-section"
  >
    <p
      className="teleinfo__dates-start-last"
    >
      <span
        className="teleinfo__dates-start-last-label"
      >
        First-last data:
      </span>
      <span
        className="teleinfo__dates-start-last-value"
      >
        HH:mm:ss(1674565891995), on yyyy/MM/dd(1674565891995)
         - 
        HH:mm:ss(1674818126026), on yyyy/MM/dd(1674818126026)
      </span>
    </p>
  </section>
</div>
`;

exports[`Teleinfo component should render correctly with notif data and power overuse 1`] = `
<div
  className="teleinfo"
>
  <section
    className="teleinfo__quick-status"
  >
    <div
      data={
        {
          "apparentPower": 250,
          "chosenFareOption": "BASE",
          "estimatedPower": 225,
          "estimatedPrices": {
            "currentDay": 5.567,
            "currentMonth": 55,
            "total": 500,
          },
          "instantIntensity": 1,
          "meta": {
            "firstDataTimestamp": 1674565891995,
            "lastUpdateTimestamp": 1674818126026,
            "unresolvedGroups": {},
          },
          "statistics": {
            "instantIntensity": {},
            "instantPower": {},
          },
          "subscribedPowerOverflowWarning": 1,
          "suppliedPower": {
            "currentDay": [
              1,
              2,
            ],
            "currentMonth": [
              10,
              20,
            ],
            "total": [
              100,
              200,
            ],
          },
        }
      }
      hearbeatTs={1674818126028}
    >
      QuickStatus component
    </div>
  </section>
  <section
    className="teleinfo__instant-section"
  >
    <p
      className="teleinfo__power"
    >
      <span
        className="teleinfo__power-value"
      >
        250
      </span>
      <span
        className="teleinfo__power-unit"
      >
        VA
      </span>
      <span
        className="teleinfo__power-value"
      >
        ~
        225
      </span>
      <span
        className="teleinfo__power-unit"
      >
        W
      </span>
    </p>
    <p
      className="teleinfo__power-stats"
    >
      <p
        className="teleinfo__power-stats-group"
      >
        <span
          className="teleinfo__power-stats-label"
        >
          Min (today/month/overall):
        </span>
        <span
          className="teleinfo__power-stats-value"
        >
          ...
        </span>
        /
        <span
          className="teleinfo__power-stats-value"
        >
          ...
        </span>
        /
        <span
          className="teleinfo__power-stats-value"
        >
          ...
        </span>
        <span
          className="teleinfo__power-stats-unit"
        >
          VA
        </span>
      </p>
      <p
        className="teleinfo__power-stats-group"
      >
        <span
          className="teleinfo__power-stats-label"
        >
          Max:
        </span>
        <span
          className="teleinfo__power-stats-value"
        >
          ...
        </span>
        /
        <span
          className="teleinfo__power-stats-value"
        >
          ...
        </span>
        /
        <span
          className="teleinfo__power-stats-value"
        >
          ...
        </span>
        <span
          className="teleinfo__power-stats-unit"
        >
          VA
        </span>
      </p>
    </p>
    <p
      className="teleinfo__intensity with-overflow"
    >
      <span
        className="teleinfo__intensity-label"
      >
        Intensity:
      </span>
      <span
        className="teleinfo__intensity-value"
      >
        1
      </span>
      <span
        className="teleinfo__intensity-unit"
      >
        A
      </span>
    </p>
    <p
      className="teleinfo__over-intensity"
    >
      <span
        className="teleinfo__over-intensity-label"
      >
        Overuse:
      </span>
      <span
        className="teleinfo__over-intensity-value"
      >
        1
      </span>
      <span
        className="teleinfo__over-intensity-unit"
      >
        A
      </span>
    </p>
    <p
      className="teleinfo__supplied"
    >
      <span
        className="teleinfo__supplied-label"
      >
        Supplied (today/month/total):
      </span>
      <span
        className="teleinfo__supplied-value"
      >
        3
      </span>
      /
      <span
        className="teleinfo__supplied-value"
      >
        30
      </span>
      /
      <span
        className="teleinfo__supplied-value"
      >
        300
      </span>
      <span
        className="teleinfo__supplied-unit"
      >
        Wh
      </span>
    </p>
    <p
      className="teleinfo__supplied-detail"
    >
      <ul
        className="teleinfo__supplied-detail-items"
      >
        <li
          className="teleinfo__supplied-detail-item"
        >
          <span
            className="teleinfo__supplied-label"
          >
            base
            :
          </span>
          <span
            className="teleinfo__supplied-value"
          >
            1
          </span>
          /
          <span
            className="teleinfo__supplied-value"
          >
            10
          </span>
          /
          <span
            className="teleinfo__supplied-value"
          >
            100
          </span>
          <span
            className="teleinfo__supplied-unit"
          >
            Wh
          </span>
        </li>
        <li
          className="teleinfo__supplied-detail-item"
        >
          <span
            className="teleinfo__supplied-label"
          >
            :
          </span>
          <span
            className="teleinfo__supplied-value"
          >
            2
          </span>
          /
          <span
            className="teleinfo__supplied-value"
          >
            20
          </span>
          /
          <span
            className="teleinfo__supplied-value"
          >
            200
          </span>
          <span
            className="teleinfo__supplied-unit"
          >
            Wh
          </span>
        </li>
      </ul>
    </p>
  </section>
  <section
    className="teleinfo__costs-section"
  >
    <p
      className="teleinfo__costs"
    >
      <span
        className="teleinfo__costs-label"
      >
        Costs (today/month/total):
      </span>
      <span
        className="teleinfo__costs-value"
      >
        ~
        5.57
      </span>
      /
      <span
        className="teleinfo__costs-value"
      >
        55
      </span>
      /
      <span
        className="teleinfo__costs-value"
      >
        500
      </span>
      <span
        className="teleinfo__costs-unit"
      >
        €
      </span>
    </p>
    <p
      className="teleinfo__costs-detail"
    >
      <ul
        className="teleinfo__costs-detail-items"
      />
    </p>
  </section>
  <section
    className="teleinfo__dates-section"
  >
    <p
      className="teleinfo__dates-start-last"
    >
      <span
        className="teleinfo__dates-start-last-label"
      >
        First-last data:
      </span>
      <span
        className="teleinfo__dates-start-last-value"
      >
        HH:mm:ss(1674565891995), on yyyy/MM/dd(1674565891995)
         - 
        HH:mm:ss(1674818126026), on yyyy/MM/dd(1674818126026)
      </span>
    </p>
  </section>
</div>
`;

exports[`Teleinfo component should render correctly with notif data and power stats 1`] = `
<div
  className="teleinfo"
>
  <section
    className="teleinfo__quick-status"
  >
    <div
      data={
        {
          "apparentPower": 250,
          "chosenFareOption": "BASE",
          "estimatedPower": 225,
          "estimatedPrices": {
            "currentDay": 5.567,
            "currentMonth": 55,
            "total": 500,
          },
          "instantIntensity": 1,
          "meta": {
            "firstDataTimestamp": 1674565891995,
            "lastUpdateTimestamp": 1674818126026,
            "unresolvedGroups": {},
          },
          "statistics": {
            "instantPower": {
              "currentDay": {
                "max": 155,
                "min": 1,
              },
              "currentMonth": {
                "max": 200,
                "min": 1.5,
              },
              "overall": {
                "max": 500,
                "min": 1.55,
              },
            },
          },
          "subscribedPowerOverflowWarning": 1,
          "suppliedPower": {
            "currentDay": [
              1,
              2,
            ],
            "currentMonth": [
              10,
              20,
            ],
            "total": [
              100,
              200,
            ],
          },
        }
      }
      hearbeatTs={1674818126028}
    >
      QuickStatus component
    </div>
  </section>
  <section
    className="teleinfo__instant-section"
  >
    <p
      className="teleinfo__power"
    >
      <span
        className="teleinfo__power-value"
      >
        250
      </span>
      <span
        className="teleinfo__power-unit"
      >
        VA
      </span>
      <span
        className="teleinfo__power-value"
      >
        ~
        225
      </span>
      <span
        className="teleinfo__power-unit"
      >
        W
      </span>
    </p>
    <p
      className="teleinfo__power-stats"
    >
      <p
        className="teleinfo__power-stats-group"
      >
        <span
          className="teleinfo__power-stats-label"
        >
          Min (today/month/overall):
        </span>
        <span
          className="teleinfo__power-stats-value"
        >
          1
        </span>
        /
        <span
          className="teleinfo__power-stats-value"
        >
          1.5
        </span>
        /
        <span
          className="teleinfo__power-stats-value"
        >
          1.55
        </span>
        <span
          className="teleinfo__power-stats-unit"
        >
          VA
        </span>
      </p>
      <p
        className="teleinfo__power-stats-group"
      >
        <span
          className="teleinfo__power-stats-label"
        >
          Max:
        </span>
        <span
          className="teleinfo__power-stats-value"
        >
          155
        </span>
        /
        <span
          className="teleinfo__power-stats-value"
        >
          200
        </span>
        /
        <span
          className="teleinfo__power-stats-value"
        >
          500
        </span>
        <span
          className="teleinfo__power-stats-unit"
        >
          VA
        </span>
      </p>
    </p>
    <p
      className="teleinfo__intensity with-overflow"
    >
      <span
        className="teleinfo__intensity-label"
      >
        Intensity:
      </span>
      <span
        className="teleinfo__intensity-value"
      >
        1
      </span>
      <span
        className="teleinfo__intensity-unit"
      >
        A
      </span>
    </p>
    <p
      className="teleinfo__over-intensity"
    >
      <span
        className="teleinfo__over-intensity-label"
      >
        Overuse:
      </span>
      <span
        className="teleinfo__over-intensity-value"
      >
        1
      </span>
      <span
        className="teleinfo__over-intensity-unit"
      >
        A
      </span>
    </p>
    <p
      className="teleinfo__supplied"
    >
      <span
        className="teleinfo__supplied-label"
      >
        Supplied (today/month/total):
      </span>
      <span
        className="teleinfo__supplied-value"
      >
        3
      </span>
      /
      <span
        className="teleinfo__supplied-value"
      >
        30
      </span>
      /
      <span
        className="teleinfo__supplied-value"
      >
        300
      </span>
      <span
        className="teleinfo__supplied-unit"
      >
        Wh
      </span>
    </p>
    <p
      className="teleinfo__supplied-detail"
    >
      <ul
        className="teleinfo__supplied-detail-items"
      >
        <li
          className="teleinfo__supplied-detail-item"
        >
          <span
            className="teleinfo__supplied-label"
          >
            base
            :
          </span>
          <span
            className="teleinfo__supplied-value"
          >
            1
          </span>
          /
          <span
            className="teleinfo__supplied-value"
          >
            10
          </span>
          /
          <span
            className="teleinfo__supplied-value"
          >
            100
          </span>
          <span
            className="teleinfo__supplied-unit"
          >
            Wh
          </span>
        </li>
        <li
          className="teleinfo__supplied-detail-item"
        >
          <span
            className="teleinfo__supplied-label"
          >
            :
          </span>
          <span
            className="teleinfo__supplied-value"
          >
            2
          </span>
          /
          <span
            className="teleinfo__supplied-value"
          >
            20
          </span>
          /
          <span
            className="teleinfo__supplied-value"
          >
            200
          </span>
          <span
            className="teleinfo__supplied-unit"
          >
            Wh
          </span>
        </li>
      </ul>
    </p>
  </section>
  <section
    className="teleinfo__costs-section"
  >
    <p
      className="teleinfo__costs"
    >
      <span
        className="teleinfo__costs-label"
      >
        Costs (today/month/total):
      </span>
      <span
        className="teleinfo__costs-value"
      >
        ~
        5.57
      </span>
      /
      <span
        className="teleinfo__costs-value"
      >
        55
      </span>
      /
      <span
        className="teleinfo__costs-value"
      >
        500
      </span>
      <span
        className="teleinfo__costs-unit"
      >
        €
      </span>
    </p>
    <p
      className="teleinfo__costs-detail"
    >
      <ul
        className="teleinfo__costs-detail-items"
      />
    </p>
  </section>
  <section
    className="teleinfo__dates-section"
  >
    <p
      className="teleinfo__dates-start-last"
    >
      <span
        className="teleinfo__dates-start-last-label"
      >
        First-last data:
      </span>
      <span
        className="teleinfo__dates-start-last-value"
      >
        HH:mm:ss(1674565891995), on yyyy/MM/dd(1674565891995)
         - 
        HH:mm:ss(1674818126026), on yyyy/MM/dd(1674818126026)
      </span>
    </p>
  </section>
</div>
`;

exports[`Teleinfo component should render correctly with notif data but without supplied power information 1`] = `
<div
  className="teleinfo"
>
  <section
    className="teleinfo__quick-status"
  >
    <div
      data={
        {
          "apparentPower": 250,
          "chosenFareOption": "BASE",
          "estimatedPower": 225,
          "estimatedPrices": {
            "currentDay": 5.567,
            "currentMonth": 55,
            "total": 500,
          },
          "instantIntensity": 1,
          "meta": {
            "firstDataTimestamp": 1674565891995,
            "lastUpdateTimestamp": 1674818126026,
            "unresolvedGroups": {},
          },
          "statistics": {
            "instantIntensity": {},
            "instantPower": {},
          },
          "suppliedPower": {},
        }
      }
      hearbeatTs={1674818126028}
    >
      QuickStatus component
    </div>
  </section>
  <section
    className="teleinfo__instant-section"
  >
    <p
      className="teleinfo__power"
    >
      <span
        className="teleinfo__power-value"
      >
        250
      </span>
      <span
        className="teleinfo__power-unit"
      >
        VA
      </span>
      <span
        className="teleinfo__power-value"
      >
        ~
        225
      </span>
      <span
        className="teleinfo__power-unit"
      >
        W
      </span>
    </p>
    <p
      className="teleinfo__power-stats"
    >
      <p
        className="teleinfo__power-stats-group"
      >
        <span
          className="teleinfo__power-stats-label"
        >
          Min (today/month/overall):
        </span>
        <span
          className="teleinfo__power-stats-value"
        >
          ...
        </span>
        /
        <span
          className="teleinfo__power-stats-value"
        >
          ...
        </span>
        /
        <span
          className="teleinfo__power-stats-value"
        >
          ...
        </span>
        <span
          className="teleinfo__power-stats-unit"
        >
          VA
        </span>
      </p>
      <p
        className="teleinfo__power-stats-group"
      >
        <span
          className="teleinfo__power-stats-label"
        >
          Max:
        </span>
        <span
          className="teleinfo__power-stats-value"
        >
          ...
        </span>
        /
        <span
          className="teleinfo__power-stats-value"
        >
          ...
        </span>
        /
        <span
          className="teleinfo__power-stats-value"
        >
          ...
        </span>
        <span
          className="teleinfo__power-stats-unit"
        >
          VA
        </span>
      </p>
    </p>
    <p
      className="teleinfo__intensity"
    >
      <span
        className="teleinfo__intensity-label"
      >
        Intensity:
      </span>
      <span
        className="teleinfo__intensity-value"
      >
        1
      </span>
      <span
        className="teleinfo__intensity-unit"
      >
        A
      </span>
    </p>
    <p
      className="teleinfo__supplied"
    >
      <span
        className="teleinfo__supplied-label"
      >
        Supplied (today/month/total):
      </span>
      <span
        className="teleinfo__supplied-value"
      >
        ...
      </span>
      /
      <span
        className="teleinfo__supplied-value"
      >
        ...
      </span>
      /
      <span
        className="teleinfo__supplied-value"
      >
        ...
      </span>
      <span
        className="teleinfo__supplied-unit"
      >
        Wh
      </span>
    </p>
    <p
      className="teleinfo__supplied-detail"
    >
      <ul
        className="teleinfo__supplied-detail-items"
      />
    </p>
  </section>
  <section
    className="teleinfo__costs-section"
  >
    <p
      className="teleinfo__costs"
    >
      <span
        className="teleinfo__costs-label"
      >
        Costs (today/month/total):
      </span>
      <span
        className="teleinfo__costs-value"
      >
        ~
        5.57
      </span>
      /
      <span
        className="teleinfo__costs-value"
      >
        55
      </span>
      /
      <span
        className="teleinfo__costs-value"
      >
        500
      </span>
      <span
        className="teleinfo__costs-unit"
      >
        €
      </span>
    </p>
    <p
      className="teleinfo__costs-detail"
    >
      <ul
        className="teleinfo__costs-detail-items"
      />
    </p>
  </section>
  <section
    className="teleinfo__dates-section"
  >
    <p
      className="teleinfo__dates-start-last"
    >
      <span
        className="teleinfo__dates-start-last-label"
      >
        First-last data:
      </span>
      <span
        className="teleinfo__dates-start-last-value"
      >
        HH:mm:ss(1674565891995), on yyyy/MM/dd(1674565891995)
         - 
        HH:mm:ss(1674818126026), on yyyy/MM/dd(1674818126026)
      </span>
    </p>
  </section>
</div>
`;

exports[`Teleinfo component should render correctly without notif data 1`] = `
<div
  className="teleinfo"
>
  <section
    className="teleinfo__quick-status"
  >
    <div>
      QuickStatus component
    </div>
  </section>
  <p
    className="teleinfo__no-data"
  >
    No data received yet.
  </p>
</div>
`;
